### PR TITLE
Port Partial Fills to Legacy Solver

### PR DESCRIPTION
**Pull request by nlordell**
This PR ports some of the logic from the co-located solver into the legacy solver. Namely:
- Managing fills between solver runs
- Computing settlements given a single order settlement and gas price estimates.

Since the math for the latter is quite tricky, I added a unit test to verify things are working as expected.

### Test Plan

Added a unit test. Full partially fillable E2E test coming up.

### CODE REVIEW

File: crates/shared/src/arguments.rs [hunk 1]
```diff
@@ -509,8 +510,10 @@ pub fn duration_from_seconds(s: &str) -> Result<Duration, ParseFloatError> {
     Ok(Duration::from_secs_f64(s.parse()?))
 }
 
-pub fn wei_from_base_unit(s: &str) -> anyhow::Result<U256> {
-    Ok(U256::from_dec_str(s)? * U256::exp10(18))
+pub fn wei_from_ether(s: &str) -> anyhow::Result<U256> {
+    let in_ether = s.parse::<BigDecimal>()?;
+    let base = BigDecimal::new(1.into(), -18);
+    number_conversions::big_decimal_to_u256(&(in_ether * base)).context("invalid Ether value")
```
Comment (vkgnosis) [inline crates/shared/src/arguments.rs:516]: Why? Fractional support?

**Review by vkgnosis — APPROVED** [review]

File: crates/shared/src/arguments.rs (hunk 1 above)
Comment (nlordell) [reply crates/shared/src/arguments.rs:516]: Exactly. We probably want this to be around 0.01 Ether or something.

Comment (nlordell) [reply crates/shared/src/arguments.rs:516]: Fixed the default I use in the command line arguments (was set to 20 Eth 🙈).

**Review by MartinquaXD — APPROVED** [review]

File: crates/solver/src/solver/single_order_solver.rs [hunk 2]
```diff
@@ -223,22 +249,117 @@ pub struct SingleOrderSettlement {
...
+                    .try_get_token_amount(
+                        &number_conversions::u256_to_big_rational(
+                            &((self.gas_estimate + gas::SETTLEMENT_OVERHEAD)
+                                * U256::from_f64_lossy(gas_price)),
+                        ),
+                        order.sell_token,
+                    )
+                    .context("failed to compute solver fee")?,
+            )
+            .context("invalid solver fee amount")?;
```
[the remark this answers is not in the document]
Comment (nlordell) [reply crates/solver/src/solver/single_order_solver.rs]: https://github.com/cowprotocol/services/pull/1393/commits/f2ce35927a7e479c0d38f4b71d81f18acabaeb84

**Pull request merged**